src/libsyntax/ext/expand.rs

```diff
@@ -367,6 +376,10 @@ pub fn expand_item_mac(it: P<ast::Item>,
         _ => fld.cx.span_bug(it.span, "invalid item macro invocation")
     });
 
+    // check here to detect use of feature-gated or invalid attributes on
+    // macro invoations since they will not be detected after macro expansion
```

Comment:
nit: move this comment to `fn check_attributes` ("this function is called to detect use ...")